components/hab/src/command/origin/delete.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2016-2018 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::api_client::Client;
+use crate::common::ui::{Status, UIWriter, UI};
+
+use crate::error::{Error, Result};
+use crate::{PRODUCT, VERSION};
+
+pub fn start(ui: &mut UI, bldr_url: &str, token: &str, origin: &str) -> Result<()> {
+    let api_client = Client::new(bldr_url, PRODUCT, VERSION, None).map_err(Error::APIClient)?;
+
+    ui.status(Status::Deleting, format!("origin {}.", origin))?;
+
+    api_client
+        .delete_origin(origin, token)
+        .map_err(Error::APIClient)?;
+
+    ui.status(Status::Deleted, format!("origin {}.", origin))?;
```

Comment:
Make this return the value of the expression and you can eliminate the `Ok(())` on L32
```suggestion
    ui.status(Status::Deleted, format!("origin {}.", origin))
```

Reply:
Removing the try here gives us a mismatch in error types because of the error type that ui.status returns. So i went with `Ok(ui.status(Status::Deleted, format!("origin {}.", origin))?)`

Reply:
My bad. I thought I had successfully built that suggestion since I'm aware of that kind of failure that the `?` operator avoids.

In this case, I think what you had is the best and I retract my suggestion. I'm not really a fan of the `Ok(…?)` form since I think it's kind of confusing. Also, when the return type is `Result<()>`, it triggers [this lint](https://rust-lang.github.io/rust-clippy/master/#unit_arg).

If you like the look of omitting the `Ok(())`, another alternative is
```rust
    ui.status(Status::Deleted, format!("origin {}.", origin))
        .map_err(Into::into)
```
That's basically doing the same thing as `?`, but without the need to re-wrap in `Ok(…)`. I kind of like the explicitness of it, but whichever form you like best is fine with me.

Reply:
:+1: